Remotion/Data/DomainObjects/Infrastructure/ObjectPersistence/RootPersistenceStrategy.cs

```diff
@@ -141,7 +148,7 @@ public virtual IEnumerable<IQueryResultRow> ExecuteCustomQuery (IQuery query)
 
       using (var storageProviderManager = CreateStorageProviderManager())
```

Comment:
StorageAccessResolver for queries is missing.
> ResolveStorageAccessForQuery
>    returns StorageAccessType.ReadWrite.
>    Is not called if the query already specyfies a ReadWrite QueryType, since the IStorageAccessResolver would not be allowed to downgrade them anyway.